cns/wireserver/net.go

```diff
@@ -49,6 +52,47 @@ func GetPrimaryInterfaceFromResult(res *GetInterfacesResult) (*InterfaceInfo, er
 	return nil, ErrNoPrimaryInterface
 }
 
+// Gets secondary interface details for swiftv2 secondary nics scenario
+func GetSecondaryInterfaceFromResult(res *GetInterfacesResult, macAddress string) (*InterfaceInfo, error) {
+	for _, i := range res.Interface {
+		// skip if primary
+		if i.IsPrimary {
+			continue
+		}
+
+		// skip if no subnets
+		if len(i.IPSubnet) == 0 {
+			continue
+		}
+
+		if macAddressesEqual(i.MacAddress, macAddress) {
+			// get the second subnet
+			s := i.IPSubnet[0]
+			gw, err := calculateGatewayIP(s.Prefix)
+			if err != nil {
+				return nil, err
+			}
+
+			secondaryIP := ""
+			for _, ip := range s.IPAddress {
+				if !ip.IsPrimary {
+					secondaryIP = ip.Address
+				}
+			}
+			var secondaryIPs []string
+			secondaryIPs = append(secondaryIPs, secondaryIP)
+
+			return &InterfaceInfo{
+				Subnet:       s.Prefix,
+				IsPrimary:    false,
+				Gateway:      gw.String(),
+				SecondaryIPs: secondaryIPs,
+			}, nil
```

Comment:
nit: maybe something like this would looks cleaner
```go
return &InterfaceInfo{
		Subnet:       s.Prefix,
		IsPrimary:    false,
		Gateway:      gw.String(),
		SecondaryIPs: []string{secondaryIP},
	}, nil
```